integrations/terraform-provider-github
# Fix go module path

### Description of the change

This PR fixes go module path per https://golang.org/ref/mod#module-path:

> If the module is released at major version 2 or higher, the module path must end with a major version suffix like /v2. This may or may not be part of the subdirectory name. 

Appends `/v4` since the current major is `v4`.

Also fixes the repository path which seems to be legacy after migrating from https://github.com/hashicorp/terraform-provider-github. 

### How has this code been tested

1. Run `make build` 
2. Verify the code compiles and builds the plugin binary
